io/subsystem/src/main/java/org/wildfly/extension/io/WorkerResourceDefinition.java

```diff
@@ -157,20 +194,52 @@ public void registerChildren(ManagementResourceRegistration resourceRegistration
     }
 
     private abstract static class AbstractWorkerAttributeHandler implements OperationStepHandler {
+        private static void populateValueFromModel(OperationContext context, ModelNode operation) {
+            final ModelNode subModel = context.readResource(PathAddress.EMPTY_ADDRESS).getModel();
+            ModelNode result = subModel.get(operation.require(ModelDescriptionConstants.NAME).asString());
+            context.getResult().set(result);
+        }
 
         public void execute(OperationContext outContext, ModelNode operation) throws OperationFailedException {
-             outContext.addStep((context, op) -> {
-                 XnioWorker worker = getXnioWorker(context);
-                 if (worker == null || worker.getMXBean() == null) {
-                     context.getResult().set(IOExtension.NO_METRICS);
-                     return;
-                 }
-                 executeWithWorker(context, op, worker);
-             }, OperationContext.Stage.RUNTIME);
-         }
+            populateValueFromModel(outContext, operation);
+            outContext.addStep((context, op) -> {
+                XnioWorker worker = getXnioWorker(context);
+                if (worker != null) {
+                    executeWithWorker(context, op, worker);
+                }
+            }, OperationContext.Stage.RUNTIME);
+        }
          abstract void executeWithWorker(OperationContext context, ModelNode operation, XnioWorker worker) throws OperationFailedException;
     }
 
+    private abstract static class WorkerWriteAttributeHandler extends AbstractWriteAttributeHandler {
+
+        @Override
+        protected boolean applyUpdateToRuntime(OperationContext context, ModelNode operation, String attributeName, ModelNode value, ModelNode currentValue, HandbackHolder handbackHolder) throws OperationFailedException {
+            XnioWorker worker = getXnioWorker(context);
+            if (worker == null) { //worker can be null if it is not started yet, it can happen when there are no dependencies to it.
+                return true;
+            }
+            try {
+                return setValue(worker, value);
+            } catch (IOException e) {
+                throw new OperationFailedException(e);
```

Comment:
I think this should be some sort of RuntimeException, unless the IOE means the user made a mistake. OFE is for user errors, not internal server faults.